community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnPageEvictionCallback.java

```diff
@@ -38,5 +38,6 @@ public void onEvict( long filePageId, Page page )
         assert removed == page :
                 "Removed unexpected page when cleaning up translation table for filePageId " + filePageId + ". " +
                         "Evicted " + page + " but removed " + removed + " from the translation table.";
+        file.setHighestEvictedTransactionId( removed.resetLastModifiedTxId() );
```

Comment:
I think we should do this before we `evictPage`, so there is no gap where "highest evicted" is lower than it should be.

Reply:
Oh, wait. We need the removed page object. In that case, I think this logic should be pushed down into `file.evictPage` so that we can capture this value before we remove the page from the translation table.

Reply:
I think this is important because while we are evicting the page, it is exclusively locked, so no one else can touch it, which in turn means that the "last modified tx id" will not change while we are evicting. The exclusive lock in turn means that as long as it remains in the translation table, no thread will initiate a page fault of that translation table entry, and therefor no thread will need the "last modified tx id" stored in the page object. So, by moving the tx id from the page object to the paged file, just before we remove the translation table entry, will ensure that no thread will miss out on any latest tx id.